roxctl/deployment/check/check.go

```diff
@@ -172,6 +175,12 @@ func (d *deploymentCheckCommand) Construct(_ []string, cmd *cobra.Command, f *pr
 }
 
 func (d *deploymentCheckCommand) Validate() error {
+	if d.cluster != "" && d.namespace == defaultNamespace {
+		d.env.Logger().WarnfLn("No custom namespace provided. Any deployments without an explicit namespace will be defaulted to \"default\". Overwrite the default with --namespace/-n [NAMESPACE]\n")
```

Comment:
Why using `WarnfLn` if:
- you do not have any formatting placeholders (e.g., `%s`)
- you add `\n` at the end

I am also not sure about the `[NAMESPACE]` placeholder. I think the message would be also clear when formulated like this:

```suggestion
		d.env.Logger().Warn("Running for 'default' namespace. Use --namespace to change the namespace.")
```

Reply:
(I suggested this message so I'm replying) 
The reason why I think including `Any deployments without an explicit namespace` makes sense is because the provided namespace does not mean that you are running the check for that namespace. It depends if the deployment yamls have the namespace set or not. 

I'm not set on the message I proposed, but I think the sentiment has to be there.

I agree with removing the `[NAMESPACE]` placeholder. 

Would you be ok with something like:

```
"Using 'default' namespace for deployments without explicit namespace. Use --namespace to change."
```

Reply:
Regarding the function, I prefer using `WarnLn` and no explicit newline at the end of the string.

Reply:
Oh, then I misunderstood the `--namespace` param - I thought it refers to a namespace on the cluster, whereas it simulates the deployments being deployed to a namespace.

Yes, then the text `"Using 'default' namespace for deployments without explicit namespace. Use --namespace to change."` makes total sense. I would also make sure that the help text for that option makes it clear what is meant for this param. If we cannot eliminate potential confusions then maybe renaming it to `--for-namespace` would be a good idea?

Reply:
> I would also make sure that the help text for that option makes it clear what is meant for this param

Agreed.

The idea here was to keep the behavior as close as possible to `kubectl -n` flag. I.e.: The property in the yaml takes precedence over the `--namespace` value, and it's defaulted to `default`. So in that regard I'd prefer to keep `-n` and `--namespace`. But lets make sure the help and warning convey that clearly. 

Ironically, kubectl help doesn't say much about this:

```
    -n, --namespace='':
	If present, the namespace scope for this CLI request
```

Reply:
There's only `WarnfLn`, no `Warn` or `WarnLn`. As for the message content I've changed it.